File: services/backend/src/main/java/com/corely/corely_backend/entity/Address.java
package com.corely.corely_backend.entity;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;
import java.util.UUID;

@Entity
@Table(name = "addresses")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Address extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    User user;

    @Column(nullable = false)
    String streetAddress;

    @Column(nullable = false)
    String ward;

    @Column(nullable = false)
    String district;

    @Column(nullable = false)
    String city;

    @Builder.Default
    Boolean isDefault = false;
}

File: services/frontend/corely-app/src/types/address.ts
export interface AddressRequest {
    streetAddress: string;
    ward: string;
    district: string;
    city: string;
    isDefault?: boolean;
}

export interface AddressResponse {
    id: string;
    streetAddress: string;
    ward: string;
    district: string;
    city: string;
    isDefault: boolean;
    createdAt: string;
    updatedAt: string;
}